### PR TITLE
Added support for Kubernetes 1.24.0

### PR DESCRIPTION
**Pull request by jala-dx**
The code requires `kubernetes.io/master` label on two places 
1. nodeSelectors
2. Tolerations

Add the `master` label right after kubeadm is inited and `before` installing the kurl addons. And this logic is added only if the current kubernetes version is >= 1.24.0

Tested for ekc-operator

```
$ kubectl get nodes --show-labels=true
NAME                                           STATUS   ROLES                  AGE   VERSION   LABELS
jalaja-k8s-v1-24-with-control-plane-labels-1   Ready    control-plane,master   26h   v1.24.0   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=jalaja-k8s-v1-24-with-control-plane-labels-1,kubernetes.io/os=linux,kurl.sh/cluster=true,node-role.kubernetes.io/control-plane=,node-role.kubernetes.io/master=,node.kubernetes.io/exclude-from-external-load-balancers=

```

#### What type of PR is this?

type::feature

#### Which issue(s) this PR fixes:

Fixes #

SC-48706

#### Does this PR introduce a user-facing change?
No

```release-note

Added support for Kubernetes 1.24.0
```

#### Does this PR require documentation?

Yes
TODO

### CODE REVIEW

**Review by garcialuis — APPROVED** [review]

File: scripts/install.sh [hunk 1]
```diff
@@ -259,6 +259,19 @@ function init() {
...
+    # once the node is up, apply the master label only on primaries
+    # get the kubernetes version
+    local kubernetes_version=$(kubectl version --short | grep -i server | awk '{ print $3 }' | sed 's/^v*//')
+    semverParse $kubernetes_version
+    KUBERNETES_CURRENT_VERSION_MAJOR="$major"
+    KUBERNETES_CURRENT_VERSION_MINOR="$minor"
+
+    if [ $KUBERNETES_CURRENT_VERSION_MAJOR -ge 1 ] && [ $KUBERNETES_CURRENT_VERSION_MINOR -ge 24 ]; then
+        local node=$(hostname | tr '[:upper:]' '[:lower:]')
+        kubectl label node "$node" node-role.kubernetes.io/master= 2>/dev/null || true
```
Comment (emosbaugh) [inline scripts/install.sh:272]:
```suggestion
        kubectl label --overwrite node "$node" node-role.kubernetes.io/master=
```

File: scripts/install.sh [hunk 2]
```diff
@@ -259,6 +259,19 @@ function init() {
...
     wait_for_nodes
+
+    # once the node is up, apply the master label only on primaries
+    # get the kubernetes version
+    local kubernetes_version=$(kubectl version --short | grep -i server | awk '{ print $3 }' | sed 's/^v*//')
+    semverParse $kubernetes_version
+    KUBERNETES_CURRENT_VERSION_MAJOR="$major"
+    KUBERNETES_CURRENT_VERSION_MINOR="$minor"
+
+    if [ $KUBERNETES_CURRENT_VERSION_MAJOR -ge 1 ] && [ $KUBERNETES_CURRENT_VERSION_MINOR -ge 24 ]; then
```
Comment (emosbaugh) [inline scripts/install.sh:270]: do you need to check this or can you just always apply this label?

**Review by emosbaugh — CHANGES_REQUESTED** [review]
How about joining a master. I think you need to add the code there as well.

File: scripts/install.sh (hunk 2 above)
Comment (jala-dx) [reply scripts/install.sh:270]: well, i would like to check this instead of applying blindly. lmk wdyt

**Comment by jala-dx** [comment]
@emosbaugh added logic for joining masters.

File: scripts/install.sh (hunk 2 above)
Comment (emosbaugh) [reply scripts/install.sh:270]: if you want to leave this code in i would prefer that you not repurpose the global variables. please either define local vars or use $major and $minor directly.

File: scripts/install.sh [hunk 3]
```diff
@@ -259,6 +259,19 @@ function init() {
...
+        local node=$(hostname | tr '[:upper:]' '[:lower:]')
+        kubectl label --overwrite node "$node" node-role.kubernetes.io/master= 2>/dev/null || true
```
Comment (emosbaugh) [inline scripts/install.sh:272]: why `2>/dev/null || true`? what do you expect to fail?

```suggestion
        kubectl label --overwrite node "$node" node-role.kubernetes.io/master=
```

File: scripts/join.sh [hunk 4]
```diff
@@ -90,6 +90,18 @@ function join() {
     if [ "$MASTER" = "1" ]; then
         exportKubeconfig
 
+        # once the node is up, apply the master label only on primaries
+        # get the kubernetes version
+        local kubernetes_version=$(kubectl version --short | grep -i server | awk '{ print $3 }' | sed 's/^v*//')
+        semverParse $kubernetes_version
+        KUBERNETES_CURRENT_VERSION_MAJOR="$major"
```
Comment (emosbaugh) [inline scripts/join.sh:97]: same as above

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: scripts/install.sh (hunk 2 above)
Comment (jala-dx) [reply scripts/install.sh]: will use $major $minor directly..

**Review by emosbaugh — COMMENTED** [review]
this looks better. can you please test on k8s 1.19,1.23 and 1.24

**Review by emosbaugh — APPROVED** [review]

**Comment by jala-dx** [comment]
> > this looks better. can you please test on k8s 1.19,1.23 and 1.24
> 
> yeah, will let you know after testing on 1.19 and 1.23 and 1.24

Kurl:   Tested on 1.19 (only with `master` label and 1.24. Going to test 1.23 on staging. 
Kots:  Built ttl image and manually edited the Deployment. And then ClusterNode, click `Add Node` to trigger the code path.
Ekco: Built ttl.sh image, manually edited the Deployment and added code to make sure it executes the new code.

```

$ kubectl get nodes --show-labels=true -A
NAME                        STATUS   ROLES    AGE   VERSION    LABELS
jalaja-kurl-1-19            Ready    master   58m   v1.19.16   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=jalaja-kurl-1-19,kubernetes.io/os=linux,kurl.sh/cluster=true,node-role.kubernetes.io/master=
jalaja-kurl-1-19-master-3   Ready    master   9m    v1.19.16   beta.kubernetes.io/arch=amd64,beta.kubernetes.io/os=linux,kubernetes.io/arch=amd64,kubernetes.io/hostname=jalaja-kurl-1-19-master-3,kubernetes.io/os=linux,kurl.sh/cluster=true,node-role.kubernetes.io/master=


$ kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.16", GitCommit:"e37e4ab4cc8dcda84f1344dda47a97bb1927d074", GitTreeState:"clean", BuildDate:"2021-10-27T16:24:44Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}

kubectl get nodes -A
NAME                        STATUS   ROLES    AGE   VERSION
jalaja-kurl-1-19            Ready    master   49m   v1.19.16
jalaja-kurl-1-19-master-3   Ready    master   23s   v1.19.16

Instance: 35.193.60.111 

```

**Pull request merged**